projects/plugins/jetpack/extensions/blocks/blogging-prompts/settings.php

```diff
@@ -29,6 +29,12 @@ function enabled_field_callback() {
  * @return void
  */
 function init() {
+	// If editor extensions are not loaded, don't show the settings.
```

Comment:
Instead of shortcircuiting here, would there be a way to do it earlier, before we even require the `settings.php` file, to save a require and the whole hooking to `admin_init` here? Given that we already have `should_load_blogging_prompts()` in `projects/plugins/jetpack/extensions/blocks/blogging-prompts/blogging-prompts.php`, maybe we could do everything there?

Reply:
ah, sorry @jeherve. The comment makes sense and good for a follow-up.

Reply:
Great point--I'll follow up shortly!

Reply:
Here's an attempt at a follow-up. Untangling this logic is surprisingly tricky! I think filtering the option wasn't actually what we needed to short-circuit the feature.

https://github.com/Automattic/jetpack/pull/27779